common/workunit/workunit.cpp

```diff
@@ -2881,19 +2881,127 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                                                 ISecManager *secmgr,
                                                 ISecUser *secuser)
     {
+        class CQueryOrFilter : public CInterface, implements IInterface
+        {
+            int fmt;
+            StringAttr name;
+            StringArray values;
+        public:
+            IMPLEMENT_IINTERFACE;
+
+            CQueryOrFilter() {};
+
+            void setName(const char* _name) { name.set(_name); };
+            void setFormat(int _fmt) { fmt = _fmt; };
+            void appendValue(const char* _value) { values.append(_value); };
+            const char* getName() { return name.get(); };
+            int getFormat() { return fmt; };
```

Comment:
This class doesn't really have an member functions.  If you have get and set functions then it would make more sense to move functionality in here - otherwise it is not really any different from making all the members public (and more code).

get functions should be const.